File: official/convex.workspace.runtime/implementations/convex_interlocking_infrastructure/fixtures/dirty/interlocking_runner_exists/convex/trains/interlocking.ts
// DIRTY fixture — coder.convex.interlocking-runner-exists.
// The InterlockingRunner class exists but ships NO resolveTrain(...) entry point, so a consumer
// declaring interlocking routes has no route-control entry. Expected: >=1 violation (missing-resolve-train).
import { TrainRunner } from "./runner";

export class InterlockingRunner {
  private readonly path: string;
  constructor(interlockingYamlPath: string) {
    this.path = interlockingYamlPath;
  }

  // BUG: no resolveTrain(...) — there is no route-control entry point at all.
  execute(action: string, inputs: Record<string, unknown>) {
    return new TrainRunner("plan/_trains/3001-solo-match-complete.yaml").execute("3001-solo-match-complete", inputs);
  }
}

File: official/atdd.extension.coder.train-interlocking/implementations/interlocking-infrastructure/fixtures/dirty_no_resolve_train/python/trains/interlocking.py
"""InterlockingRunner — DIRTY (no resolve_train) fixture.

The class exists but exposes no resolve_train(...) entry point — there is no way to select an
admissible route before delegating (core afokapu/atdd#1251). It does delegate cleanly otherwise.
"""
from trains.runner import TrainRunner


class InterlockingRunner:
    def __init__(self, interlocking_yaml_path):
        self._path = interlocking_yaml_path

    def execute(self, action, inputs, state=None, _train_runner=TrainRunner):
        # BUG: jumps straight to a hardcoded train with no resolve_train route selection.
        return _train_runner("plan/_trains/3001-solo-match-complete.yaml").execute(
            "3001-solo-match-complete", inputs
        )
